engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/KafkaAvroSourceFactory.scala

```diff
@@ -14,20 +14,22 @@ import pl.touk.nussknacker.engine.avro.typed.AvroSchemaTypeDefinitionExtractor
 import pl.touk.nussknacker.engine.kafka.KafkaSourceFactory._
 import pl.touk.nussknacker.engine.kafka._
 import pl.touk.nussknacker.engine.kafka.serialization.DeserializationSchemaFactory
+import pl.touk.nussknacker.engine.util.namespaces.ObjectNamingProvider
 
 class KafkaAvroSourceFactory[T: TypeInformation](config: KafkaConfig,
                                                  schemaFactory: DeserializationSchemaFactory[T],
                                                  schemaRegistryClientFactory: SchemaRegistryClientFactory,
                                                  timestampAssigner: Option[TimestampAssigner[T]],
                                                  formatKey: Boolean = false)
-  extends KafkaSourceFactory[T](config, schemaFactory, timestampAssigner, TestParsingUtils.newLineSplit) {
+  extends KafkaSourceFactory[T](config, schemaFactory, timestampAssigner,
+    TestParsingUtils.newLineSplit, ObjectNamingProvider) {
 
   override protected def createSource(processMetaData: MetaData,
                                       topics: List[String],
                                       schema: KafkaDeserializationSchema[T]): KafkaSource = {
     val schemaRegistryClient = schemaRegistryClientFactory.createSchemaRegistryClient(config)
     new KafkaSource(consumerGroupId = processMetaData.id, topics, schema,
-      Some(AvroToJsonFormatter(schemaRegistryClient, topics.head, formatKey)))
+      Some(AvroToJsonFormatter(schemaRegistryClient, topics.head, formatKey)), ObjectNamingProvider)
```

Comment:
why we use default here? And in all places other than ProcessConfigCreator?

Reply:
It's not necessarily default - `ObjectNamingProvider` uses `ScalaServiceLoader`, which will load the default `ObjectNaming` implementation `DefaultObjectNaming` only if there is no other implementation of `ObjectNaming` provided. I wanted to pass the provider to `KafkaSource` and `KafkaSink` only for testing purposes. I can move it higher up the tree than to `KafkaSourceFactory[T]` and `KafkaSinkFactory` as it is currently, but at some point `ObjectNamingProvider` will have to be passed anyway - so I suppose the question here is where is it reasonable to stop.